kore/src/Kore/Strategies/Goal.hs

```diff
@@ -979,14 +986,29 @@ removalPatterns
   , isFunctionPattern destTerm
   = do
     unifiedConfigs <-
-        simplifyConditionalTermToOr
-            sideCondition
-            (mkIn configSort configTerm destTerm)
+        mkIn configSort configTerm destTerm
+        & Pattern.fromTermLike
+        & Pattern.simplify sideCondition
     if isBottom unifiedConfigs
         then return OrPattern.top
         else do
+            let unifiedConditions =
+                    fmap Conditional.withoutTerm unifiedConfigs
+            -- TODO (thomas.tuegel): Move this up to avoid repeated calls.
+            destinationConditions <-
+                Conditional.withoutTerm destination
+                & Condition.simplifyCondition sideCondition
+                & OrCondition.observeAllT
+            remainderConditions <-
+                simplifyEvaluatedMultiPredicate
+                    sideCondition
+                    (MultiAnd.make
+                        [ unifiedConditions
+                        , destinationConditions
+                        ]
+                    )
             let remainderPatterns =
-                    fmap remainderPattern unifiedConfigs
+                    fmap Pattern.fromCondition_ remainderConditions
```

Comment:
Ah, so we also needed to first fully simplify the destination and all the resulting conditions together before calling `simplifyEvaluated` for `Exists` and `Not`, right?

Reply:
Yes, that's right: the simplifiers for `Exists` and `Not` assume that their arguments are fully simplified, but they don't check. What makes it even harder is that _usually_ the patterns are fully simplified at this point, but in a few cases they need extra function or simplification rules to be applied.